aws/resource_aws_autoscaling_group_test.go

```diff
@@ -124,6 +124,7 @@ func TestAccAWSAutoScalingGroup_basic(t *testing.T) {
 					resource.TestCheckResourceAttr("aws_autoscaling_group.bar", "termination_policies.0", "OldestInstance"),
 					resource.TestCheckResourceAttr("aws_autoscaling_group.bar", "termination_policies.1", "ClosestToNextInstanceHour"),
 					resource.TestCheckResourceAttr("aws_autoscaling_group.bar", "vpc_zone_identifier.#", "0"),
+					resource.TestCheckResourceAttr("aws_autoscaling_group.bar", "max_instance_lifetime", "864000"),
```

Comment:
When the schema is fixed to be optional instead of required, the expected value here is 0 when its unconfigured:

```
--- FAIL: TestAccAWSAutoScalingGroup_basic (173.02s)
    testing.go:635: Step 0 error: Check failed: Check 32/32 error: aws_autoscaling_group.bar: Attribute 'max_instance_lifetime' expected "864000", got "0"
```

```suggestion
					resource.TestCheckResourceAttr("aws_autoscaling_group.bar", "max_instance_lifetime", "0"),
```